src/thread.rs

```diff
@@ -52,29 +63,49 @@ impl Thread {
         })
     }
 
-    /// Destroys the given thread waiting for it to become ready to destroy.
-    pub fn destroy(self) {
-        debug!(
-            "Destroying SPDK thread {}... {:p}",
-            self.name(),
-            self.as_ptr()
-        );
+    /// Marks thread as exiting.
+    pub fn exit(&self) {
+        info!("Exiting SPDK thread: {:?}", self);
 
+        let _g = CurrentThreadGuard::new();
+        self.set_current();
         unsafe {
-            spdk_set_thread(self.as_ptr());
+            spdk_thread_exit(self.as_ptr());
+        }
+    }
+
+    /// Marks a thread as exiting, and waits until it exits by polling it.
+    pub fn wait_exit(&self) {
+        info!("Waiting SPDK thread to exit: {:?}", self);
+
+        let _g = CurrentThreadGuard::new();
 
-            // set that we *want* to exit, but we have not exited yet
+        self.set_current();
+
+        unsafe {
             spdk_thread_exit(self.as_ptr());
 
             // now wait until the thread is actually exited the internal
             // state is updated by spdk_thread_poll()
             while !spdk_thread_is_exited(self.as_ptr()) {
                 spdk_thread_poll(self.as_ptr(), 0, 0);
             }
-            spdk_thread_destroy(self.as_ptr());
         }
+    }
+
+    /// Destroys a thread, freeing all of its resources.
+    /// Only an exited thread can be safely destroyed, so client code
+    /// must ensure the thread has exited before destroying it.
+    pub fn destroy(self) {
+        info!("Destroying SPDK thread: {:?}", self);
```

Comment:
Looks like previously these were debug rather than info. Will info happen too frequently and pollute the logs?

Reply:
Ok, will change to debug.